docs/contracts/visibility-and-getters.rst

```diff
@@ -186,39 +186,74 @@ to write a function, for example:
         }
     }
 
-Now you can use ``getArray()`` to retrieve the entire array, instead of
-``myArray(i)``, which returns a single element per call.
+If you have a ``public`` state variable of mapping type. The getter function
+returns the value associated with the key passed as an argument.
+
+.. code-block:: solidity
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity >=0.4.16 <0.9.0;
+
+    contract MappingExample {
+        // public state variable
+        mapping(uint => uint) public myMap;
+
+        // Getter function generated by the compiler
+        /*
+        function myMap(uint i) public view returns (uint) {
+            return myMap[i];
+        }
+        */
+    }
 
-The next example is more complex:
+When you declare a public state variable of a struct type, the generated getter function
+returns each member of the struct as separate elements within a tuple,
+rather than the struct as a single object in memory.
+The members returned appear in the order they are declared in the struct, provided they
+are not omitted. The mapping and arrays (with the exception of byte arrays)
+in the struct are omitted because there is no good way to
+select individual array members or provide a key for the mapping
+Additionally, if all struct members are omitted, no getter function will be generated.
```

Comment:
We should clarify that this is a compilation error. The compiler won't just silently skip the getter.